File: Selenium/Activities/Activity17.java
/*Open a new browser to https://training-support.net/webelements/selects
Get the title of the page and print it to the console.
On the Multi Select:
Select the "HTML" option using the visible text.
Select the 4th, 5th and 6th options using the index.
Select the "Node" option using the value.
Deselect the 5th option using index.*/

package training;

import java.util.List;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class Activity17 {
	public static void main(String[] args) {
		//Create instance of firefox driver
		WebDriver driver = new FirefoxDriver();
		//Open the web page
		driver.get("https://training-support.net/webelements/selects");
		//Print the title of the page
		System.out.println("Title of the page: " + driver.getTitle());
		//Find the drop down
		WebElement selectElement = driver.findElement((By.cssSelector("select.h-80")));
		//Pass the webelement to select the object
		Select multiSelect = new Select(selectElement);
		
		//Select "html" using the visible text
		multiSelect.selectByVisibleText("HTML");
		//Select 4th, 5th and 6th index options
		for (int i=3; i<=5; i++) {
			multiSelect.selectByIndex(i);
		}
		//SElect "Node" using value attribute
		multiSelect.selectByValue("nodejs");
		
		//Print the selected options
		List<WebElement> selectedOptions = multiSelect.getAllSelectedOptions();
		System.out.println("Selected options are, ");
		for (WebElement option: selectedOptions) {
			System.out.println(option.getText());	
		}
		//Deselect the 5th index option
		multiSelect.deselectByIndex(5);
		//Print the selected options
		selectedOptions = multiSelect.getAllSelectedOptions();
		for (WebElement option: selectedOptions) {
			System.out.println(option.getText());	
		}
		//Close the browser
		driver.quit();
		
	}
}

File: Selenium/Activities/Activity17.py
"""Open a new browser to https://training-support.net/webelements/selects
Get the title of the page and print it to the console.
On the Multi Select:
Select the "HTML" option using the visible text.
Select the 4th, 5th and 6th options using the index.
Select the "Node" option using the value.
Deselect the 5th option using index."""

#Import the webdriver
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select

#Start the driver
with webdriver.Firefox() as driver:
    #Open the browser
    driver.get("https://training-support.net/webelements/selects")
    #Print the title of the page
    print("Title of the page: ", driver.title) 
    #Find the dropdown
    selectElement = driver.find_element(By.CSS_SELECTOR, "select.h-80")
    #pass the webelement to the select object
    multiSelect = Select(selectElement)

    #Select html using visible text
    multiSelect.select_by_visible_text("HTML")
    #Print the selected option
    print("Selected option by visible text: ", multiSelect.first_selected_option.text)

    #Select 4th, 5th and 6th index options
    for i in range(3,5):
        multiSelect.select_by_index(i)
    #Select "Node" using value attribute
    multiSelect.select_by_value("nodejs")
    
    #Print the selected option
    selectedOptions = multiSelect.all_selected_options
    print("Selected options are, ")
    for option in selectedOptions:
        print(option.text)
    
    # Deselect the 5th index option
    multiSelect.deselect_by_index(4)
    # Print the selected options
    selectedOptions = multiSelect.all_selected_options
    print("Selected options are: ")
    for option in selectedOptions:
        print(option.text)
